Itsdangerous replacement with PyJWT

gerwout opened the pull request:
Newer versions of Itsdangerous (i.e. 2.1.1) no longer have support for JWT encoding and/or decoding (https://github.com/pallets/itsdangerous/blob/d1c85670cce70d81f9949619434daf8c0b9cd37e/src/itsdangerous/jws.py#L23-L30). This is a breaking change for flask-oidc. This PR replaces the itsdangerous functionality with PyJWT.

gerwout committed:
Replaced itsdangerous JWT encoding and decoding with PyJWT library
Itsdangerous no longer has JWT encoding or decoding functionality with
newer versions (i.e. 2.1.1). This will completely break flask_oidc

gerwout committed:
- removed itsdangerous, added PyJWT

billsteve commented:
Very nice.

palewire commented:
It would be great to get this merged and released.

ecederstrand committed:
Remove submodule to allow pip install using git

Commit:
Let Makefile handle themes git repo. Fixes #152

Commit:
Only check existence of submodule if it's needed

gerwout committed:
Merge pull request #1 from ecederstrand/patch-1

Allow pip install using git

ecederstrand commented on requirements.txt:
```diff
@@ -1,4 +1,4 @@
 Flask
-itsdangerous
+PyJWT
```
This change needs to also go into `install_requires` in setup.py

tapanhalani commented:
If everything looks good, please merge this PR asap. We use flask_oidc with superset which breaks due to this.

khteh commented:
This project / repo needs a new owner.

ecederstrand committed:
Reuse requirements.txt

This removes duplication and pulls in the itsdangerous -> PyJWT change into setup.pp

ecederstrand committed:
Pull in requirements.txt into MANIFEST

This allow installs from the source package to work

rvhonorato commented:
Any idea when this will be merged?

gerwout committed:
Merge pull request #2 from ecederstrand/patch-1

Reuse requirements.txt to complete PyJWT switch

gerwout commented:
@ecederstrand sorry for my late response, I overlooked your merge request, but I have merged your changes now as well. Would be nice if this PR can now be merged into master

Panagiss commented:
@puiterwijk This needs a merge ASAP!

benpocklingtonhes commented:
@puiterwijk Please can you merge this

gautierrog commented:
Any chance to see this MR merged?

frozenpandaman commented:
Still waiting…

rvhonorato commented:
still waiting :)

benjaminknox commented:
Please merge!

abompard commented:
Hello everyone, this version of flask-oidc is abandoned. It has been forked and is now maintained at https://github.com/fedora-infra/flask-oidc/ . The new version has a few breaking changes though, please read the docs if you upgrade.
